app/controllers/products_controller.rb

```diff
@@ -0,0 +1,9 @@
+class ProductsController < ApplicationController
+  def index
+    @products = Product.published.default_order
+  end
+
+  def show
+    @product = Product.find(params[:id])
```

Comment:
```suggestion
    @product = Product.find(params[:id])
```
これ、非公開のやつをid直接指定したら見えちゃうのでは？